packages/php-wasm/compile/php/php_wasm.c

```diff
@@ -3,6 +3,26 @@
  *
  * This file abstracts the entire PHP API with the minimal set
  * of functions required to run PHP code from JavaScript.
+ *
+ * !! NOTE FOR FUTURE MAINTAINERS !!
+ *
+ * When debugging memory errors, it's important to understand the behavior of the stack in the Emscripten environment:
+ *
+ * - The stack does not grow automatically. If the stack runs out of space, it will not be expanded automatically, unlike in traditional C programs built with compilers like GCC or Clang.
```

Comment:
let's get some confirmation on this, as I was under the impression that the stack does not traditionally grow in C programs built with GCC or clang. maybe there's some kind of compiler option I'm not familiar with, but I think this will typically blow a stack long before we run out of memory.

in fact, the following program "blows the stack" through recursion after only 261,539 recursive calls, which is well below the amount of available memory on my 32 GB machine

```c
#include <stdio.h>

int recurse(int c) {
	return c == 0 ? 0 : recurse(c - 1);
}

int main() {
	int n = 261539 + 1; // fails at the +1
	printf("Recursed %d times\n", recurse( n ));
	return 1;
}
```

Reply:
@dmsnell I've removed the text `unlike in traditional C programs built with compilers like GCC or Clang`.